src/test/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/FindOperationTest.java

```diff
@@ -1511,10 +1521,223 @@ public void findAllSort() throws Exception {
+                                  DocumentId.fromString("doc2"))),
+                          Values.of(UUID.randomUUID()),
+                          Values.of(doc2),
+                          Values.NULL,
+                          Values.NULL,
+                          Values.NULL,
+                          Values.NULL,
+                          Values.of(1672531300000L)),
+                      List.of(
+                          Values.of(
+                              CustomValueSerializers.getDocumentIdValue(
+                                  DocumentId.fromString("doc1"))),
+                          Values.of(UUID.randomUUID()),
+                          Values.of(doc1),
+                          Values.NULL,
+                          Values.NULL,
+                          Values.NULL,
+                          Values.NULL,
+                          Values.of(1672531200000L)),
+                      List.of(
+                          Values.of(
+                              CustomValueSerializers.getDocumentIdValue(
+                                  DocumentId.fromString("doc3"))),
+                          Values.of(UUID.randomUUID()),
+                          Values.of(doc3),
+                          Values.NULL,
+                          Values.NULL,
+                          Values.NULL,
+                          Values.NULL,
+                          Values.of(1672531400000L)),
+                      List.of(
+                          Values.of(
+                              CustomValueSerializers.getDocumentIdValue(
+                                  DocumentId.fromString("doc5"))),
+                          Values.of(UUID.randomUUID()),
+                          Values.of(doc5),
+                          Values.NULL,
+                          Values.NULL,
+                          Values.NULL,
+                          Values.NULL,
+                          Values.of(1672531600000L))));
+
+      FindOperation operation =
+          new FindOperation(
+              COMMAND_CONTEXT,
+              List.of(),
+              DocumentProjector.identityProjector(),
+              null,
+              5,
+              20,
+              ReadType.SORTED_DOCUMENT,
+              objectMapper,
+              List.of(new FindOperation.OrderBy("sort_date", true)),
+              0,
+              20);
+
+      Supplier<CommandResult> execute =
+          operation
+              .execute(queryExecutor)
+              .subscribe()
+              .withSubscriber(UniAssertSubscriber.create())
+              .awaitItem()
+              .getItem();
+
+      // assert query execution
+      candidatesAssert.assertExecuteCount().isOne();
+
+      // then result
+      CommandResult result = execute.get();
+      assertThat(result.data().docs())
+          .hasSize(5)
+          .isEqualTo(
+              List.of(
+                  objectMapper.readTree(doc1),
```

Comment:
I'd recommend adding another query to this test that sorts in reverse order, and asserts that sorting by `new FindOperation.OrderBy("sort_date", false)` returns doc6, doc5, doc4, doc3, doc2. This test as written might succeed even if no sorting was done, because the documents happen to be in ascending order by `sort_date`.